pkg/api/validation/validation.go

```diff
@@ -432,6 +435,17 @@ func validateGlusterfs(glusterfs *api.GlusterfsVolumeSource) errs.ValidationErro
 	return allErrs
 }
 
+func validateMetadataVolumeSource(metadata *api.MetadataVolumeSource) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	for _, metadataFile := range metadata.Items {
+		if len(metadataFile.Name) == 0 {
```

Comment:
Should we validated that the file name is a valid relative path? Ie, I assume you want to be able to do `some/dir/filename` but we should not allow `some/dir/../../../../../../../etc/passwd`.